### PR TITLE
tz metadata

### PR DESCRIPTION
**Pull request by ice-orion**

### CODE REVIEW

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 1]
```diff
@@ -51,100 +53,66 @@ export function* updateDeviceMetadataSaga(action: Action) {
...
-        buildId,
-        bootloader: bootloader,
-        codename,
-        installerPackageName,
-        pushNotificationToken,
-        userId,
-        deviceUniqueId,
+        pushNotificationToken: hasPushPermissions ? messaging().getToken() : '',
+        tz:
+          (timezoneOffset > 0 ? '+' : '-') +
```
Comment (ice-ares) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts:105]: >=0

Comment (ice-ares) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:105]: ‘>=0’

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 2]
```diff
@@ -51,100 +53,66 @@ export function* updateDeviceMetadataSaga(action: Action) {
...
         tablet: DeviceInfo.isTablet(),
-        pinOrFingerprintSet,
-        emulator,
-        firstInstallTime:
-          firstInstallTime === -1 ? dayjs().valueOf() : firstInstallTime,
-        lastUpdateTime:
-          lastUpdateTime === -1 ? dayjs().valueOf() : lastUpdateTime,
         systemName: DeviceInfo.getSystemName(),
         systemVersion: DeviceInfo.getSystemVersion(),
-        apiLevel: apiLevel === -1 ? 999 : apiLevel,
```
Comment (ice-ares) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts:134]: keep this logic

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts (hunk 1 above)
Comment (ice-orion) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts]: done

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts (hunk 2 above)
Comment (ice-orion) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:134]: done, removed by mistake

**Review by ice-ares — DISMISSED** [review]

**Review by ice-hades — APPROVED** [review]

**Comment by ice-hades** [comment]
@ice-orion @ice-ares maybe we can send here only updates. I mean now we collect and send all the data every time. But most of the data doesn't change, like brand, tablet, firstInstallTime etc.

**Comment by ice-ares** [comment]
> @ice-orion @ice-ares maybe we can send here only updates. I mean now we collect and send all the data every time. But most of the data doesn't change, like brand, tablet, firstInstallTime etc.

thats exactly what we want, to send it every X minutes

**Review by ice-atlas — CHANGES_REQUESTED** [review]

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 3]
```diff
@@ -19,25 +22,24 @@ import {
 import {isPermissionGrantedSelector} from '@store/modules/Permissions/selectors';
 import {getErrorMessage} from '@utils/errors';
 import DeviceInfo from 'react-native-device-info';
-import {call, put, select} from 'redux-saga/effects';
+import {all, call, put, select} from 'redux-saga/effects';
 
 type Action = ReturnType<
```
[the remark this answers is not in the document]
Comment (ice-orion) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:27]: we don't need to differ payload here

[the remark this answers is not in the document]
Comment (ice-atlas) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:27]: this is not type safe 
```
action.payload ?? {}
```

**Pull request merged**